### PR TITLE
NAS-114013 / 22.02.1 / Explicitly restart/stop serial-getty service based on user specified settings

### PR DESCRIPTION
**Pull request by sonicaj**
This PR introduces changes to disable serial-getty services by default. Now moving on, we explicitly enable/disable the service on serial port which user has specified for serial console.

### CODE REVIEW

File: src/middlewared/middlewared/etc_files/ttys_config.py [hunk 1]
```diff
@@ -1,19 +1,13 @@
-import os
-import signal
-import subprocess
+from middlewared.utils import run
```
Comment (themylogin) [inline src/middlewared/middlewared/etc_files/ttys_config.py:1]: This does not generate any config files anymore, can we drop `ttys` pseudo-service and move it to a middleware plugin?

File: src/middlewared/middlewared/plugins/system.py [hunk 2]
```diff
@@ -360,23 +360,48 @@ async def do_update(self, data):
...
+                    )
+                    if cp.returncode:
+                        self.logger.error('Failed to %r serialconsole: %r', action, cp.stderr.decode())
+            if (
                 original_data['serialspeed'] != config_data['serialspeed'] or
                 original_data['serialport'] != config_data['serialport']
             ):
-                generate_grub = True
+                if original_data['serialport'] != config_data['serialport']:
+                    commands = [
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/system.py:386]: Where is this variable used?

**Review by themylogin — CHANGES_REQUESTED** [review]
Nice!

Can we request test VM to have two serial ports and then try testing changing this value?

File: src/middlewared/middlewared/plugins/system.py (hunk 2 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/system.py:386]: Which one ?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/system.py:386]: `commands`

**Review by themylogin — CHANGES_REQUESTED** [review]

**Review by themylogin — APPROVED** [review]
Super cool, thank you!

**Pull request merged**